### PR TITLE
feat(menu): open to space

### PR DESCRIPTION
**Pull request by AriLFrankel**
Open menu to where there is space for it. If there is not room for it below open vertically. If there is not room to the right open to the left.

### CODE REVIEW

File: src/components/Menu/Menu.js [hunk 1]
```diff
@@ -1,65 +1,86 @@
...
+    const overBottom = top + menuRect.height > window.innerHeight;
+    const overRight = left + menuRect.width > window.innerWidth;
+    this.menu.style.top = `${anchorRect.top -
+      (overBottom ? menuRect.height - anchorRect.height : 0)}px`;
+    this.menu.style.left = `${anchorRect.left -
+      (overRight ? menuRect.width - anchorRect.width : 0)}px`;
+  }, 0);
 
   render() {
+    this.recalculatePosition();
```
Comment (martahj) [inline src/components/Menu/Menu.js]: [Q] Why is this in render? It's not changing state so it's not triggering a rerender, but still feels a little less intuitive than calculating it on scroll

Comment (AriLFrankel) [reply src/components/Menu/Menu.js]: @martahj If we only trigger `recalculatePosition` on scroll or resize events then the menu will not be placed correctly when it is first opened before any of those events take place

Comment (AriLFrankel) [reply src/components/Menu/Menu.js]: I like that a lot!

**Comment by martahj** [comment]
Looks pretty good to me!

In lieu of a Now app, please verbally verify that this is working locally 😅

**Comment by AriLFrankel** [comment]
I hereby verify that it works on my machine lol

**Pull request merged**